tools/scripts/build_projects.py

```diff
@@ -0,0 +1,76 @@
+#!/bin/python
+
+import argparse
+import json
+import os
+import multiprocessing
+import sys
+
+TGREEN =  '\033[32m' # Green Text	
+TBLUE =  '\033[34m' # Green Text	
+TRED =  '\033[31m' # Red Text	
+TWHITE = '\033[39m' #Withe text
+
+description_help='''Build all noos projects
+Examples:\n
+    >python build_projects.py ..\.. aducm3029
+	>python build_projects.py /home/user/noos xilinx
+'''
+
+def parse_input():
+    parser = argparse.ArgumentParser(description=description_help,\
+			    formatter_class=argparse.RawTextHelpFormatter)
+    parser.add_argument('noos_location', help="Path to noos location")
+    parser.add_argument('export_dir', help="Path where to save files")
+    parser.add_argument('platform', help="Platform to build for",
+				choices=['aducm3029', 'xilinx', 'stm32', 'altera'])
+    args = parser.parse_args()
+
+    return (args.noos_location, args.platform, args.export_dir)
+
+def run_cmd(cmd):
+	log_file = 'log.txt'
+	print(cmd)
+	sys.stdout.flush()
+	err = os.system(cmd + ' > %s 2>&1' % log_file)
+	if (err != 0):
+		print(TGREEN + "Error" + TWHITE)
+		print("See log:")
+		os.system("cat %s" % log_file)
+
+def to_blue(str):
+	return TBLUE + str + TWHITE
+
+def main():
+	create_dir_cmd = "test -d {0} || mkdir -p {0}"
+	(noos, platform, export_dir) = parse_input()
+	projets = os.path.join(noos,'projects')
+	run_cmd(create_dir_cmd.format(export_dir))
+	for project in os.listdir(projets):
+		projet_dir = os.path.join(projets, project)
+		build_file = os.path.join(projet_dir, 'builds.json')
+		if os.path.isfile(build_file):
+			fp = open(build_file)
+			configs = json.loads(fp.read())
+			for (cur_platform, config) in configs.items():
+				if (cur_platform == platform):
+					platform_export_dir = os.path.join(export_dir, platform)
+					run_cmd(create_dir_cmd.format(platform_export_dir))
+					for (build_name, flags) in config.items():
+						print("Runing build %s on project %s for platform %s" %
+							(to_blue(build_name),
+							to_blue(project),
+							to_blue(cur_platform)))
+						build_dir_name = 'build_%s_%s' % (cur_platform, build_name)
+						binary = os.path.join(build_dir_name, "%s_%s.elf" % (project, build_name))
+						export_file = os.path.join(projet_dir, binary)
+						cmd = 'make -C %s %s BUILD_DIR_NAME=%s\
+ LOCAL_BUILD=n LINK_SRCS=n BINARY=%s VERBOSE=y -j%d ' % (projet_dir, flags, build_dir_name, binary, multiprocessing.cpu_count() - 1)
+						run_cmd(cmd + 'ra')
+						if (cur_platform == 'aducm3029'):
```

Comment:
I don't think this kind of stuff should belong here. If you need to do aducm specific actions, you should add an optional postbuild: or postprocessing: rule to build system, and each platform may or may not implement it.

This script should contain the logic to build any project - period, it shouldn't know anything specific.

Reply:
just the replacement of .elf to .hex is keept.